client/state/ui/comments/reducer.js

```diff
@@ -0,0 +1,46 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get, isUndefined, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { COMMENTS_PER_PAGE } from 'my-sites/comments/constants';
+import { COMMENTS_QUERY_UPDATE } from 'state/action-types';
+import { combineReducers, keyedReducer } from 'state/utils';
+
+const deepUpdateComments = ( state, comments, { offset, postId, status } ) => {
+	const parent = postId || 'site';
+	const pageNumber = offset / COMMENTS_PER_PAGE + 1;
+	const commentIds = map( comments, 'ID' );
+
+	const parentObject = get( state, parent, {} );
+	const statusObject = get( parentObject, status, {} );
+
+	return {
+		...state,
+		[ parent ]: {
+			...parentObject,
+			[ status ]: {
+				...statusObject,
+				[ pageNumber ]: commentIds,
```

Comment:
Note that we'll need to modify the structure if we add search later

Reply:
Well, we'll need to change `status` into something like `${ status }?s=${ searchQuery }`.